### ISSUE TITLE
HiddenFieldCheck: parameter shadowing top-level field not flagged in compact source files (JEP 512)

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#20222]
Parent Issue #19971

I have read check documentation: https://checkstyle.org/checks/coding/hiddenfield.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="HiddenField"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > InputHiddenFieldCompact.java <<'EOF'
String name = "default";
int count = 0;
void process(String name, int count) {
    System.out.println(name + ": " + count);
}
void main() { process("test", 42); }
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.5.0-SNAPSHOT-all.jar -c config.xml InputHiddenFieldCompact.java

Starting audit...
Audit done.
```

Same code inside an ordinary class:

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > OrdinaryHiddenField.java <<'EOF'
public class OrdinaryHiddenField {
    String name = "default";
    int count = 0;
    void process(String name, int count) {
        System.out.println(name + ": " + count);
    }
}
EOF

vivek@Viveks-MacBook-Air checkstyle % java $RUN_LOCALE -jar target/checkstyle-13.5.0-SNAPSHOT-all.jar -c config.xml OrdinaryHiddenField.java

Starting audit...
[ERROR] OrdinaryHiddenField.java:4:25: 'name' hides a field. [HiddenField]
[ERROR] OrdinaryHiddenField.java:4:35: 'count' hides a field. [HiddenField]
Audit done.
Checkstyle ends with 2 errors.
```

---

**Describe what you expect in detail.**

HiddenFieldCheck must detect a method parameter that shadows a top-level field in a JEP 512 compact source file. The check builds its `FieldFrame` by visiting `CLASS_DEF`, `ENUM_DEF`, and `RECORD_DEF` tokens and collecting field declarations from their bodies. A compact source file's implicit class is represented by none of these tokens, so top-level fields are never registered in any `FieldFrame`, and parameter shadowing goes undetected.

Expected on the compact input: 2 violations (`'name' hides a field.` and `'count' hides a field.`), identical to the ordinary class above.

**Comment by vivek-0509** [comment]
@Carbon14-48 Hey if you want you can help us with this issue

**Comment by Carbon14-48** [comment]
> [@Carbon14-48](https://github.com/Carbon14-48) Hey if you want you can help us with this issue

sure !
